### DISCUSSION TITLE
2FA for macOS app

### DISCUSSION THREAD

**Discussion by nkamenar** [dani-garcia/vaultwarden#4396] [Q&A]
I have been using Bitwarden for a while with a self hosted Vaultwarden instance. I have my account secured to use 2FA with a couple yubikeys which has been working great until I just tried to log in with my new MacBook Pro laptop. When I downloaded the Bitwarden app from the App Store and tried to log in I am getting the error from the image attached. Is it really not possible to use a Yubikey with Bitwarden on macOS? If not what 2FA methods are supported? Any assistance is much appreciated.
![](https://community.bitwarden.com/uploads/default/original/3X/7/8/7812cc3292d4c7536a0f595506821fe9bedc8d99.png)

**Comment by stefan0xC** [comment]
It's not yet supported by Bitwarden: https://community.bitwarden.com/t/fido2-support-for-macos-and-linux-desktop-client/11867/79
Cf. https://github.com/bitwarden/clients/pull/6375 and https://github.com/bitwarden/clients/pull/7188

The [authenticator provider](https://bitwarden.com/help/setup-two-step-login-authenticator/) should be supported and probably [email as well](https://bitwarden.com/help/setup-two-step-login-email/).
